include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,1550 @@
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            Assert(r < 1, ExcInternalError());
+            for (unsigned int c=0; c<n_cols; ++c)
+              {
+                Assert(c < 1, ExcInternalError());
+                out(r,c) = s;
+              }
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<0,dim,Number> &s)
+      {
+        return unroll_to_matrix(s.operator const Number &());
+      }
+
+
+      template<int dim, typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<1,dim,Number> &v)
+      {
+        FullMatrix<Number> out (v.n_independent_components,1);
+        const unsigned int n_rows = out.m();
+        const unsigned int n_cols = out.n();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices = internal::indices_from_component<dim>(r,false);
+            Assert(indices.first < dim, ExcInternalError());
+            const unsigned int &i = indices.first;
+
+            for (unsigned int c=0; c<n_cols; ++c)
+              {
+                Assert(c < 1, ExcInternalError());
+                out(r,c) = v[i];
+              }
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const Tensor<2,dim,Number> &t)
+      {
+        FullMatrix<Number> out (dim,dim);
+        const unsigned int n_rows = out.m();
+        const unsigned int n_cols = out.n();
+        for (unsigned int r=0; r<n_rows; ++r)
+          {
+            const std::pair<unsigned int,unsigned int> indices_i = internal::indices_from_component<dim>(r,false);
+            Assert(indices_i.first < dim, ExcInternalError());
+            Assert(indices_i.second < dim, ExcInternalError());
+            const unsigned int &i = indices_i.first;
+
+            for (unsigned int c=0; c<n_cols; ++c)
+              {
+                const std::pair<unsigned int,unsigned int> indices_j = internal::indices_from_component<dim>(c,false);
+                Assert(indices_j.first < dim, ExcInternalError());
+                Assert(indices_j.second < dim, ExcInternalError());
+                const unsigned int &j = indices_j.second;
+
+                out(r,c) = t[i][j];
+              }
+          }
+        return out;
+      }
+
+
+      template<int dim, typename Number>
+      FullMatrix<Number>
+      unroll_to_matrix (const SymmetricTensor<2,dim,Number> &st)
+      {
+        return unroll_to_matrix(Tensor<2,dim,Number>(st));
+      }
+
+
+      namespace internal
+      {
+        namespace
+        {
+          template<typename TensorType>
+          struct is_rank_1_tensor : std::false_type
+          {};
+
+          template<int dim, typename Number>
+          struct is_rank_1_tensor<Tensor<1,dim,Number> >
+            : std::true_type
+          {};
```

Comment:
is this still necessary? can't you just query `TensorType::rank == 1` now that you added the static member variable?

Reply:
Yes, you're right. I've removed these now, but note that I've kept the helper struct in to test for a symmetric tensor - this is because the `SubTensor` type sent into the rank-3 tensor conversion functions need not necessarily have the same number type as that of the input / output tensor.